lang/src/lib.rs

```diff
@@ -193,6 +194,26 @@ pub trait Event: AnchorSerialize + AnchorDeserialize + Discriminator {
     fn data(&self) -> Vec<u8>;
 }
 
+/// `emit!()` for an event that is emitted through a CPI
+#[macro_export]
+macro_rules! emit_cpi {
+    ($event:expr, $program_info:expr) => {{
+        let program_info: &AccountInfo = $program_info;
+        _emit_cpi_data!($event);
+        crate::_emit_cpi_invoke(__ix_data, program_info)?;
+    }};
+}
+
+pub fn _emit_cpi_invoke(ix_data: Vec<u8>, program: &AccountInfo) -> Result<()> {
```

Comment:
can you put this inside the __private module below?

Reply:
@armaniferrante fixed